Kconfig.zephyr

```diff
@@ -738,6 +738,24 @@ endmenu
 
 menu "Boot Options"
 
+config APPLICATION_VERSION_SUPPORTED
```

Comment:
Isn't this unnecessary? It seems like the right dependency ordering would be to make the version option `depend on BOOTLOADER_MCUBOOT` instead since it is imgtool specific.

Reply:
As above, it is so that other systems that can have an application version can just select this symbol. Another candidate in addition to those mentioned above is binary descriptors https://github.com/zephyrproject-rtos/zephyr/pull/54464